addon/content/new-tab-variation.js

```diff
@@ -21,6 +21,16 @@ class TrackingProtectionStudy {
     addMessageListener("TrackingStudy:UpdateContent", this);
     addMessageListener("TrackingStudy:ShuttingDown", this);
     addMessageListener("TrackingStudy:Uninstalling", this);
+
+    this.initTimer();
+  }
+
+  initTimer() {
+    this.openingTime = Math.floor(Date.now() / 1000);
+    this.contentWindow.addEventListener("beforeunload", ()=> {
+      sendAsyncMessage("TrackingStudy::NewTabOpenTime",
+        Math.floor(Date.now() / 1000) - this.openingTime);
```

Comment:
This method produces wildly inaccurate times for the newtab page being open! I have no idea why as the math seems fine:
![wildlyinaccuratetimes](https://user-images.githubusercontent.com/17437436/36566443-5a1b18b6-17d8-11e8-9a63-ea0b022fb6b4.gif)

Reply:
Also, there should be a corresponding `removeEventListener` for this as well.

Reply:
I will add the removeEvenetListener

Reply:
I figured out why the timer is broken! Thanks to Mardak of the Activity Stream team.

**Tl;dr: Set the pref `"browser.newtab.preload"` to `false` at the beginning of the study and reset it at the end of the study. Kamyar, can you add this change to your list of action items?**

_A longer explanation:_
I thought perhaps there was a problem with `Date.now()` and I tried to use a relative measure of time instead using `setInterval` and incrementing a counter every second. What I discovered was that the frame script's `init` method (where the timer is initialized) is actually being called twice the first time I visit `about:newtab`.

Per Mardak:
>  preloading is only for getting the next about:newtab ready. the very first new tab is not preloaded though
the first about:newtab is not preloaded, opening a new tab will trigger a regular load and a preload for the next one
when you open a second about:newtab, the preloaded one was already loaded and it now appears. then a new preload is triggered for the 3rd time
there is a perf impact as without preloading, the browser spends time loading javascript, react rendering, waiting for messages, etc

Sure enough, when I tried this out by setting that pref in this branch, the timer works!